contracts/Compass.vy

```diff
@@ -173,51 +233,58 @@ def update_valset(consensus: Consensus, new_valset: Valset):
 # This makes calls to contracts that execute arbitrary logic
 # message_id is to prevent replay attack and every message_id can be used only once
 @external
-def submit_logic_call(consensus: Consensus, args: LogicCallArgs, message_id: uint256, deadline: uint256):
-    assert block.timestamp <= deadline, "Timeout"
+def submit_logic_call(consensus: Consensus, args: LogicCallArgs, fee_args: FeeArgs, message_id: uint256, deadline: uint256):
+    FeeManager(FEE_MANAGER).transfer_fees(fee_args, msg.sender)
```

Comment:
@webelf101 @taariq 

At the moment, the relayer would receive the full reimbursement (total estimates gas cost) for every call before any security checks (deadline, checkpoint, etc...). This needs to happen after `check_validator_signatures` for sure, otherwise I can just start sending messages and get a lot of reimbursements.

Reply:
@webelf101 This is still open

Reply:
No worries. It is fine. Even this worked though, if the following operation fails, the tx reverted and the above work isn't pushed in block.